packages/trader/src/AppV2/Hooks/useOrderDetails.ts

```diff
@@ -0,0 +1,163 @@
+import {
+    CONTRACT_TYPES,
+    TContractInfo,
+    addComma,
+    getDurationPeriod,
+    getDurationTime,
+    getDurationUnitText,
+    getGrowthRatePercentage,
+    isAccumulatorContract,
+    isResetContract,
+} from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { getBarrierValue } from 'App/Components/Elements/PositionsDrawer/helpers';
+
+// Contains all key values that are used more than once in different transform objects
+const getCommonFields = (data: TContractInfo) => {
+    const { tick_count, tick_passed, contract_type } = data;
+    const ticks_label = Number(tick_count) < 2 ? localize('tick') : localize('ticks');
+    const ticks_duration_text = isAccumulatorContract(contract_type)
+        ? `${tick_passed}/${tick_count} ${localize('ticks')}`
+        : `${tick_count} ${ticks_label}`;
+
+    return {
+        [localize('Reference ID')]: [
+            data.transaction_ids?.buy ? `${data.transaction_ids.buy} (Buy)` : '',
+            data.transaction_ids?.sell ? `${data.transaction_ids.sell} (Sell)` : '',
+        ],
+        [localize('Stake')]: data.buy_price ? `${data.buy_price.toFixed(2)} ${data.currency}` : '',
+        [localize('Duration')]:
+            Number(tick_count) > 0
+                ? ticks_duration_text
+                : `${getDurationTime(data) ?? ''} ${getDurationUnitText(getDurationPeriod(data)) ?? ''}`,
+        [localize('Payout per point')]: data.display_number_of_contracts ? `${data.display_number_of_contracts}` : '',
+    };
+};
+
+// For Multiplier
+const transformMultiplierData = (data: TContractInfo) => {
+    const commonFields = getCommonFields(data);
+    return {
+        [localize('Reference ID')]: commonFields[localize('Reference ID')],
+        [localize('Multiplier')]: data.multiplier ? `x${data.multiplier}` : '',
+        [localize('Stake')]: commonFields[localize('Stake')],
+        [localize('Commission')]: data.commission ? `${data.commission} ${data.currency}` : '',
+        [localize('Take Profit')]: data.limit_order?.take_profit?.order_amount
+            ? `${data.limit_order.take_profit.order_amount.toFixed(2)} ${data.currency}`
+            : localize('Not set'),
+        [localize('Stop loss')]: data.limit_order?.stop_loss?.order_amount
+            ? `${data.limit_order.stop_loss.order_amount.toFixed(2)} ${data.currency}`
+            : localize('Not set'),
+        [localize('Stop out level')]: data.limit_order?.stop_out?.order_amount
+            ? `${data.limit_order.stop_out.order_amount.toFixed(2)} ${data.currency}`
+            : '',
+    };
+};
+
+// For Rise
+const transformRiseData = (data: TContractInfo) => {
```

Comment:
If it's used also for Higher/Lower, maybe we should mark it in function name somehow?